The job **`x86_64-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/10443652067/job/28917312079) [(plain)](https://github.com/rust-lang-ci/rust/commit/20629389b9ae42c17ba954151d2a6d236ac235cb/checks/28917312079/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
test [crashes] tests\crashes\129166.rs ... ok
test [crashes] tests\crashes\129205.rs ... ok
test [crashes] tests\crashes\129127.rs ... ok
test [crashes] tests\crashes\129150.rs ... ok
2024-08-18T21:36:51.415744Z ERROR compiletest::runtest: fatal error, panic: "crashtest no longer crashes/triggers ICE, horray! Please give it a meaningful name, add a doc-comment to the start of the test explaining why it exists and move it to tests/ui or wherever you see fit. Adding 'Fixes #<issueNr>' to your PR description ensures that the corresponding ticket is auto-closed upon merge."
test [crashes] tests\crashes\129209.rs ... ok
test [crashes] tests\crashes\129214.rs ... ok
test [crashes] tests\crashes\129219.rs ... ok
test [crashes] tests\crashes\34127.rs ... ok
---
failures:

---- [crashes] tests\crashes\129216.rs stdout ----

error: crashtest no longer crashes/triggers ICE, horray! Please give it a meaningful name, add a doc-comment to the start of the test explaining why it exists and move it to tests/ui or wherever you see fit. Adding 'Fixes #<issueNr>' to your PR description ensures that the corresponding ticket is auto-closed upon merge.
thread '[crashes] tests\crashes\129216.rs' panicked at src\tools\compiletest\src\runtest.rs:368:18:
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
failures:
    [crashes] tests\crashes\129216.rs

test result: FAILED. 231 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 7.05s

Some tests failed in compiletest suite=crashes mode=crashes host=x86_64-pc-windows-msvc target=x86_64-pc-windows-msvc
Build completed unsuccessfully in 0:43:10
make: *** [Makefile:102: ci-msvc-ps1] Error 1
  network time: Sun, 18 Aug 2024 21:36:54 GMT
##[error]Process completed with exit code 2.
Post job cleanup.
[command]"C:\Program Files\Git\bin\git.exe" version
```

</details>

--- a/tests/crashes/129150.rs
+++ b/tests/crashes/129150.rs
@@ -1,0 +1,7 @@
+//@ known-bug: rust-lang/rust#129150
+//@ only-x86_64
+use std::arch::x86_64::_mm_blend_ps;
+
+pub fn main() {
+     _mm_blend_ps(1, 2, &const {} );
+}

--- a/tests/crashes/129166.rs
+++ b/tests/crashes/129166.rs
@@ -1,0 +1,7 @@
+//@ known-bug: rust-lang/rust#129166
+
+fn main() {
+    #[cfg_eval]
+    #[cfg]
+    0
+}

--- a/tests/crashes/129205.rs
+++ b/tests/crashes/129205.rs
@@ -1,0 +1,5 @@
+//@ known-bug: rust-lang/rust#129205
+
+fn x<T: Copy>() {
+    T::try_from();
+}

--- a/tests/crashes/129209.rs
+++ b/tests/crashes/129209.rs
@@ -1,0 +1,11 @@
+//@ known-bug: rust-lang/rust#129209
+
+impl<
+        const N: usize = {
+            static || {
+                Foo([0; X]);
+            }
+        },
+    > PartialEq for True
+{
+}

--- a/tests/crashes/129214.rs
+++ b/tests/crashes/129214.rs
@@ -1,0 +1,30 @@
+//@ known-bug: rust-lang/rust#129214
+//@ compile-flags: -Zvalidate-mir -Copt-level=3 --crate-type=lib
+
+trait to_str {}
+
+trait map<T> {
+    fn map<U, F>(&self, f: F) -> Vec<U>
+    where
+        F: FnMut(&Box<usize>) -> U;
+}
+impl<T> map<T> for Vec<T> {
+    fn map<U, F>(&self, mut f: F) -> Vec<U>
+    where
+        F: FnMut(&T) -> U,
+    {
+        let mut r = Vec::new();
+        for i in self {
+            r.push(f(i));
+        }
+        r
+    }
+}
+
+fn foo<U, T: map<U>>(x: T) -> Vec<String> {
+    x.map(|_e| "hi".to_string())
+}
+
+pub fn main() {
+    assert_eq!(foo(vec![1]), ["hi".to_string()]);
+}

--- a/tests/crashes/129216.rs
+++ b/tests/crashes/129216.rs
@@ -1,0 +1,12 @@
+//@ known-bug: rust-lang/rust#129216
+//@ only-linux
+
+trait Mirror {
+    type Assoc;
+}
+
+struct Foo;
+
+fn main() {
+    <Foo as Mirror>::Assoc::new();
+}

--- a/tests/crashes/129219.rs
+++ b/tests/crashes/129219.rs
@@ -1,0 +1,26 @@
+//@ known-bug: rust-lang/rust#129219
+//@ compile-flags: -Zmir-opt-level=5 -Zvalidate-mir --edition=2018
+
+use core::marker::Unsize;
+
+pub trait CastTo<T: ?Sized>: Unsize<T> {}
+
+impl<T: ?Sized, U: ?Sized> CastTo<T> for U {}
+
+impl<T: ?Sized> Cast for T {}
+pub trait Cast {
+    fn cast<T: ?Sized>(&self) -> &T
+    where
+        Self: CastTo<T>,
+    {
+        self
+    }
+}
+
+pub trait Foo {}
+impl Foo for [i32; 0] {}
+
+fn main() {
+    let x: &dyn Foo = &[];
+    let x = x.cast::<[i32]>();
+}

